helpers/peers_manager.js

```diff
@@ -52,22 +51,15 @@ PeersManager.prototype.add = function(peer) {
 		return false;
 	}
 	this.peers[peer.string] = peer;
-	if (!this.addressToNonceMap[peer.string]) {
+	if (!this.addressToNonceMap[peer.string] && !peer.socket) {
 		// Create client WS connection to peer
-		connect(peer, this.logger, () => {
-			// Upon disconnection, if the peer is still in the list,
-			// set the peer state to disconnected.
-			// The peer will only be removed when the master process receives the
-			// command to do so from the worker process; the worker process decides
-			// when the master process should remove a peer.
-			var lostPeer = this.peers[peer.string];
-			if (lostPeer) {
-				lostPeer.state = Peer.STATE.DISCONNECTED;
-			}
-		});
+		connect(peer, this.logger);
+	} else {
+		// Reconnect existing socket
+		peer.socket.connect();
 	}
-	this.addressToNonceMap[peer.string] = peer.nonce;
 	if (peer.nonce) {
```

Comment:
we don't need a check for `peer.string`? I think consistent check again both `peer.string` and `peer.nonce` will ensure consistency.